sdk/metricsadvisor/azure-ai-metricsadvisor/azure/ai/metricsadvisor/_metrics_advisor_client.py

```diff
@@ -401,25 +431,40 @@ def list_anomalies_for_detection_configuration(self, detection_configuration_id,
         :raises ~azure.core.exceptions.HttpResponseError:
         """
 
-        skip = kwargs.pop('skip', None)
-        filter_condition = kwargs.pop('filter', None)
-        converted_start_time = convert_datetime(start_time)
-        converted_end_time = convert_datetime(end_time)
-        detection_anomaly_result_query = DetectionAnomalyResultQuery(
-            start_time=converted_start_time,
-            end_time=converted_end_time,
-            filter=filter_condition,
-        )
-
-        return self._client.get_anomalies_by_anomaly_detection_configuration(  # type: ignore
-            configuration_id=detection_configuration_id,
-            skip=skip,
-            body=detection_anomaly_result_query,
-            cls=lambda objs: [DataPointAnomaly._from_generated(x) for x in objs],
-            **kwargs)
+    @distributed_trace
+    def list_anomalies(self, **kwargs):
+        # type: (Any) -> ItemPaged[DataPointAnomaly]
+        """Query anomalies under a specific alert or detection configuration.
+
+        :keyword str alert_configuration_id: anomaly alert configuration unique id.
+        :keyword str alert_id: alert id.
+        :keyword str detection_configuration_id: anomaly detection configuration unique id.
+        :keyword Union[str, datetime.datetime] start_time: start time filter under chosen time mode.
+        :keyword Union[str, datetime.datetime] end_time: end time filter under chosen time mode.
+        :keyword int skip:
+        :return: Anomalies under a specific alert or detection configuration.
+        :rtype: ~azure.core.async_paging.AsyncItemPaged[~azure.ai.metricsadvisor.models.DataPointAnomaly]
+        :raises ~azure.core.exceptions.HttpResponseError:
+        """
+        alert_configuration_id = kwargs.get('alert_configuration_id', None)
+        alert_id = kwargs.get('alert_id', None)
+        detection_configuration_id = kwargs.get('detection_configuration_id', None)
+        start_time = kwargs.get('start_time', None)
+        end_time = kwargs.get('end_time', None)
+        if detection_configuration_id:
+            if alert_configuration_id or alert_id:
+                raise TypeError(
+                    'Specify either "detection_configuration_id" or "alert_configuration_id" and "alert_id"'
```

Comment:
nit: since start_time and end_time are required, should the error message be...

`'Specify either "detection_configuration_id, start_time, end_time" or "alert_configuration_id", "alert_id"'`

That might hopefully stop a user from hitting two TypeErrors in a row.

Reply:
This is a good question. I thought about this but felt it was too length to have the entire list. I am leaning towards shorter and more accurate errors.

If user species both detection_id & alert_id, we only complain they should not input both.

If user misses start_time for detection_id, we only tell them they need to specify start_time as well.

Reply:
that sounds good to me